src/services/infra/InfraService.ts

```diff
@@ -324,32 +340,104 @@ export default class InfraService {
+          }
+
+          const successEmailDetails = {
+            subject: `Launch site ${message.repoName} SUCCESS`,
+            body: `<p>Isomer site ${message.repoName} was launched successfully.</p>
+            <p>You may now visit your live website. <a href="${message.primaryDomainSource}">${message.primaryDomainSource}</a> should be accessible within a few minutes.</p>
+            <p>This email was sent from the Isomer CMS backend.</p>`,
+          }
+
+          const failureEmailDetails = {
+            subject: `Launch site ${message.repoName} FAILURE`,
+            body: `<p>Isomer site ${message.repoName} was not launched successfully.</p>
+            <p>Error: ${message.statusMetadata}</p>
+            <p>This email was sent from the Isomer CMS backend.</p>
+            `,
+          }
+
+          let emailDetails: { subject: string; body: string }
+          if (message.status) {
+            emailDetails = successEmailDetails
+          } else {
+            updateSuccessSiteLaunchParams = {
+              id: site.id,
+              siteStatus: SiteStatus.Initialized,
+              jobStatus: JobStatus.Failed,
+            }
+            emailDetails = failureEmailDetails
+          }
+          await this.sitesService.update(updateSuccessSiteLaunchParams)
+          await mailer.sendMail(
+            message.requestorEmail,
+            emailDetails.subject,
+            emailDetails.body
+          )
+
+          let params
+          if (message.status === SITE_LAUNCH_LAMBDA_STATUS.SUCCESS) {
+            params = {
               id: site.id,
               siteStatus: SiteStatus.Launched,
               jobStatus: JobStatus.Running,
             }
-            await this.sitesService.update(updateSuccessSiteLaunchParams)
+            emailDetails.subject = `Launch site ${message.repoName} SUCCESS`
+            emailDetails.body = `<p>Isomer site ${message.repoName} was launched successfully.</p>
+          <p>You may now visit your live website. <a href="${message.primaryDomainSource}">${message.primaryDomainSource}</a> should be accessible within a few minutes.</p>
+          <p>This email was sent from the Isomer CMS backend.</p>`
+          } else {
+            params = { id: site.id, jobStatus: JobStatus.Failed }
+            emailDetails.subject = `Launch site ${message.repoName} FAILURE`
+            emailDetails.body = `<p>Isomer site ${message.repoName} was not launched successfully.</p>
+          <p>Error: ${message.statusMetadata}</p>
+          <p>This email was sent from the Isomer CMS backend.</p>
+          `
           }
-        })
-      )
-    } catch (e) {
-      logger.error(e)
-    }
+          await mailer.sendMail(
+            message.agencyEmail,
+            emailDetails.subject,
+            emailDetails.body
+          )
+          await mailer.sendMail(
+            message.requestorEmail,
+            emailDetails.subject,
+            emailDetails.body
```

Comment:
all of these have too much assignment, which makes it abit confusing to follow - perhaps we should consider shifting these out into helper methods?

Reply:
8f6bb37 opps, there were some redundant code, removed!